src/util/sss_ini.c

```diff
@@ -180,14 +179,29 @@ static int sss_ini_access_check(struct sss_ini *self)
         flags |= INI_ACCESS_CHECK_UID | INI_ACCESS_CHECK_GID;
     }
 
-    ret = ini_config_access_check(self->file,
-                                  flags,
-                                  geteuid(),
-                                  getegid(),
-                                  S_IRUSR, /* r**------ */
-                                  ALLPERMS & ~(S_IWUSR|S_IXUSR));
+    if (geteuid() == 0)
+        return EOK;
 
-    return ret;
+    /* There is no legit reason to have any write/modify permissions on the config file. */
+    const char *name = ini_config_get_filename(self->file);
+    if (name != NULL) {
+        if (access(name, W_OK) == 0) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "%s is writable to sssd, which it should not be.\n", name);
+            return EACCES;
+        } else if (access(name, R_OK) != 0) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "%s is not readable.\n", name);
+            return EACCES;
+        }
+    }
+
+    /* Also ensure we are barred from calling chmod. */
```

Comment:
What do you mean - "barred"?
If somebody writes sssd.conf from scratch under 'root', it is still required to chown to sssd group, right?